### PR TITLE
PROD-909 Disable the registration page mapping on activity settings save

### LINKED ISSUE #1553: Registration pages are self assigning when Activity settings is changed
**Describe the bug**
When setting BuddyBoss as Registration Form in BuddyBoss > Settings > Registration, this opens up pages in Registration Pages in BuddyBoss > Pages > Registration Pages. The initial selection would be "select a page", but when you change something in BuddyBoss > Settings > Activity, the Registration pages are self assigning

**To Reproduce**
1. Go to BuddyBoss > Settings > Registration. Enable BuddyBoss Registration in Registration Form
2. Go to BuddyBoss > Pages > Registration Pages, see these are all in "Select a page" choices (don't select anything, keep as is)
3. Go to BuddyBoss > Settings > Activity. Change something, like  enable/disable: Automatically check for new activity posts
4. Go back to BuddyBoss > Pages > Registration Pages, see error. The pages are already assigned

**Expected behavior**
The Registration Pages must not be changed

**Screenshots**
https://drive.google.com/file/d/1mk8gvKnvOJEWFSw_dZxCGqd6QEdtLtCu/view?usp=sharing
original issue by customer: https://vimeo.com/461016238/0a652425e8

**Support ticket links**
https://secure.helpscout.net/conversation/1228780073/83847

**Jira issue** : [PROD-909]

[PROD-909]: https://buddyboss.atlassian.net/browse/PROD-909?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by surajkrsingh**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #1553 .

### How to test the changes in this Pull Request:

1. Enable Registration from general settings
2. Move on the page settings and unset the Register Form page from the registration pages section
3. Move on the activity settings page and save it
4. Back on page settings and check the Register Form page option
5. The option will not auto-assign on the last save action

### Proof Screenshots or Video
https://www.awesomescreenshot.com/video/5971551?key=4d3ef947971fe6b038d3f80c9c200e3a

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Disable the registration page mapping on activity settings save action

### CODE REVIEW

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@surajkrsingh Need to add the condition for this https://github.com/buddyboss/buddyboss-platform/pull/3154/files#diff-83f9ddd1b991122f9c36c95cfbc26d8b648c59dc04c83e9517602dc30fe10766R714-R728 pages also.

**Comment by surajkrsingh** [comment]
@chetansatasiya Changes updated

Condition added for the term and policy pages also.

File: src/bp-core/bp-core-functions.php [hunk 1]
```diff
@@ -660,12 +660,14 @@ function bp_core_get_directory_pages() {
...
- *                           are kept, and the gaps filled in with new pages. Default: 'keep'.
+ * @param array  $components   Components to create pages for.
+ * @param string $existing     'delete' if you want to delete existing page mappings
+ *                             and replace with new ones. Otherwise existing page mappings
+ *                             are kept, and the gaps filled in with new pages. Default: 'keep'.
+ * @param bool   $map_register Whether or not mapping the registration pages.
+ *                             Default: true
  */
-function bp_core_add_page_mappings( $components, $existing = 'keep' ) {
+function bp_core_add_page_mappings( $components, $existing = 'keep', $map_register = true ) {
```
Comment (dipskakadiya) [inline src/bp-core/bp-core-functions.php:670]: @surajkrsingh I think the `map_register` name is not correct this flag is not only for registration we checking for privacy policy also I think we choose an appropriate name something like `page_mapping`

File: src/bp-core/admin/settings/bp-admin-setting-activity.php [hunk 2]
```diff
@@ -68,7 +68,9 @@ public function settings_save() {
 
 		$bp->active_components = $active_components;
 		bp_core_install( $bp->active_components );
-		bp_core_add_page_mappings( $bp->active_components );
+
+		// Mapping the component pages in page settings except registration pages.
+		bp_core_add_page_mappings( $bp->active_components, 'keep', false );
```
Comment (dipskakadiya) [inline src/bp-core/admin/settings/bp-admin-setting-activity.php:73]: @surajkrsingh Can you check how many time this function call in the product?  I think page creation logic should call only once. 
Look like this is not the correct way. Ping me we will discuss it on call.

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

File: src/bp-core/bp-core-functions.php (hunk 1 above)
Comment (surajkrsingh) [reply src/bp-core/bp-core-functions.php:670]: @dipskakadiya
With this param,  we are just checking the registration pages.
image

So, that it's is `$map_register` 
Let me know if still need to rename the param.

File: src/bp-core/admin/settings/bp-admin-setting-activity.php (hunk 2 above)
Comment (surajkrsingh) [reply src/bp-core/admin/settings/bp-admin-setting-activity.php:73]: @dipskakadiya 
This is called around 13 places in the product(on a different case and different functions) except for test cases.
Total 21 times including test cases.

**Review by dipskakadiya — APPROVED** [review]

**Review by chetansatasiya — APPROVED** [review]

**Review by chetansatasiya — APPROVED** [review]

**Pull request merged**